packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -148,18 +148,22 @@ export type RenderState = {
   hoistableChunks: Array<Chunk | PrecomputedChunk>,
 
   // Flushing queues for Resource dependencies
-  preconnects: Set<PreconnectResource>,
-  fontPreloads: Set<PreloadResource>,
-  highImagePreloads: Set<PreloadResource>,
+  preconnects: Set<Resource>,
+  fontPreloads: Set<Resource>,
+  highImagePreloads: Set<Resource>,
   // usedImagePreloads: Set<PreloadResource>,
-  precedences: Map<string, Map<string, StyleResource>>,
-  stylePrecedences: Map<string, StyleTagResource>,
-  bootstrapScripts: Set<PreloadResource>,
-  scripts: Set<ScriptResource>,
-  bulkPreloads: Set<PreloadResource>,
+  precedences: Map<string, PrecedenceQueue>,
+  bootstrapScripts: Set<Resource>,
+  scripts: Set<Resource>,
+  bulkPreloads: Set<Resource>,
```

Comment:
should this just be `preloads.unknown`?

Reply:
no, image is currently modeled as a known resource type that also sometimes flushes in this queue. I would say that the queue names should probably be about priority and not tied directly to the resource types. I do want to rename though, just haven't though of the right naming scheme and since this will potentially need to change with upcoming work on Suspense everywhere I think I'll just leave it as is for now

Reply:
`preloads.other`?

Reply:
It should be grouped with the others though, right?

Reply:
I've been thinking about the `preloads.images` as a working state to reason about "upgrading" preloads to other kinds of resources (or in the case of images just upgrading its priority). However There really isn't a conflict in using them as flushing queues. Once consequence though is that it bakes in a sort of precedence because we can only empty these working queues one at a time whereas script, style, image, and arbitrary preloads today can flush in discovery order in the bulkPreloads queue.

I don't think it makes a huge difference either way and I kind of like separating the "keep track of our preload states so we can act on them later" and "here is the order I want to flush things"